app/jobs/wallet_transactions/create_job.rb

```diff
@@ -4,9 +4,13 @@ module WalletTransactions
   class CreateJob < ApplicationJob
     queue_as 'high_priority'
 
-    def perform(organization_id:, params:)
+    def perform(organization_id:, params:, new_wallet: false)
       organization = Organization.find(organization_id)
       WalletTransactions::CreateService.call(organization:, params:)
```

Comment:
It looks like we are not testing the result of the service in this job:
```suggestion
      WalletTransactions::CreateService.call!(organization:, params:)
```

Any reason not to add the webhook logic inside the service? It feels a bit weird to add business logic in the job class

Reply:
Yeah, it's not obvious from the diff but when a wallet is created, there is often an initial amount, either paid or offered (granted) and [this first balance is added to the wallet in a wallet_transaction](https://github.com/getlago/lago-api/pull/3133/files#diff-12e7d1a7ede943998b91fa6f76498376d687459fc3a0ec6ded0019f35c1b6477R42-R43) created async.

If I dispatch the webhook in `Wallets::CreateService` the balance may be zero or be correct, depending on which jobs is executed first. I think it's best to send the webhook when the initial amount is correct.

That's why I tried to explain in the PR description 😝

Reply:
Ok, but I don't see the diff with calling the `SendWebhookJob` as the last step of the `WalletTransactions::CreateService.call`. 
If it is async anyway, it should not change much thing :thinking:

Reply:
> Any reason not to add the webhook logic inside the service?
I was surprised too 🤷‍♂️  I'll merge your suggestion

Ok I think I understand, you'd prefer to have the `new_wallet` carried over the WalletTransaction::CreateService so the logic live in the service, not in the job? Actually I agree 👍 👍

Reply:
Actually I have reconsidered and updated the description